mlflow/types/utils.py

```diff
@@ -205,6 +205,19 @@ def _infer_schema(data: Any) -> Schema:
     return schema
 
 
+def _infer_inference_schema(params: Dict[str, Any], optional: bool) -> InferenceSchema:
```

Comment:
actually, this is the specific place where `optional` and `is_required` don't line up, meaning that you always have to negate one to translate between modules

Reply:
I got that point, however, it's a private method so when naming it I tried to maximize usage. Parameters are optional by default and in the MLflow typing system, types use the property `optional` for that. So if you do nothing you get optional parameters in the signature. Adding `require_parameters=True` instead of `optional_parameters=False` looked like a better programming model to me. Does it make sense?

Reply:
require_parameters does seem clearer from a user perspective. We can see if anyone else flags it on review?